sycl/test-e2e/Matrix/joint_matrix_hip_copy.hpp

```diff
@@ -0,0 +1,118 @@
+#include <sycl/sycl.hpp>
+
+#include <cstddef>
+#include <cstdint>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using sycl::ext::oneapi::bfloat16;
+
+template <typename, size_t M, size_t N> struct input_limit {
+  static constexpr int value = M * N;
+};
+
+template <> struct input_limit<int8_t, 16, 16> {
+  static constexpr auto value = 128;
+};
+
+template <> struct input_limit<int8_t, 32, 32> {
+  static constexpr auto value = 128;
+};
+
+template <typename InType, typename OutType, size_t M, size_t N, size_t K,
+          layout OutLayout>
+void hip_matrix_copy() {
+  InType A[M * K];
+  InType B[K * N];
+  OutType C[M * N];
+  OutType D[M * N];
+  OutType E[M * N];
+
+  for (auto i = 0; i < M * K; ++i) {
+    A[i] = i % input_limit<InType, M, N>::value;
+  }
+
+  for (auto i = 0; i < K * N; ++i) {
+    B[i] = i % input_limit<InType, M, N>::value;
+  }
+
+  for (auto i = 0; i < M * N; ++i) {
+    D[i] = 0;
+    C[i] = i;
+    if (OutLayout == layout::row_major)
+      E[i] = i;
+    else
+      E[(i % N) * M + int(i / M)] = i;
+  }
+
+  try {
+    auto defaultQueue = sycl::queue{};
+
+    auto bufA = sycl::buffer{A, sycl::range{M * K}};
+    auto bufB = sycl::buffer{B, sycl::range{K * N}};
+    auto bufC = sycl::buffer{C, sycl::range{M * N}};
+    auto bufD = sycl::buffer{D, sycl::range{M * N}};
+
+    defaultQueue
+        .submit([&](sycl::handler &cgh) {
+          sycl::accessor accA{bufA, cgh, sycl::read_only};
+          sycl::accessor accB{bufB, cgh, sycl::read_only};
+          sycl::accessor accC{bufC, cgh, sycl::read_only};
+          sycl::accessor accD{bufD, cgh, sycl::write_only};
+
+          cgh.parallel_for(
+              sycl::nd_range<2>{{4, 16}, {4, 16}}, [=](sycl::nd_item<2> idx) {
+                auto sg = idx.get_sub_group();
+                joint_matrix<sub_group, OutType, use::accumulator, M, N> sub_c,
+                    sub_c_copy{};
+                joint_matrix<sub_group, InType, use::b, K, N, layout::row_major>
+                    sub_b{}, sub_b_copy{};
+                joint_matrix<sub_group, InType, use::a, M, K, layout::col_major>
+                    sub_a, sub_a_copy{};
+
+                joint_matrix_copy(sg, sub_c, sub_c_copy);
```

Comment:
sub_c, sub_a, sub_b have not been initialized to anything.
So the copy here is not testing anything.
I would first load into sub_a, sub_b, sub_c and then call the  copy.

Reply:
I

> sub_c, sub_a, sub_b have not been initialized to anything. So the copy here is not testing anything. I would first load into sub_a, sub_b, sub_c and then call the copy.

Thanks @dkhaldi 
The intension of the test was to copy after load, not sure why I made this mistake.

Reply:
> sub_c, sub_a, sub_b have not been initialized to anything. So the copy here is not testing anything. I would first load into sub_a, sub_b, sub_c and then call the copy.

Thanks @dkhaldi 
Thanks for spotting that. The intention of the test was to `copy` after `load` as you say.